integr8ly/integreatly-operator
# MGDAPI-5833 Removed remaining OO code

# Issue link
JIRA: [MGDAPI-5833](https://issues.redhat.com/browse/MGDAPI-5833)

# What
This PR removes the remaining references to OO including the OO uninstallation logic which can now be removed since RHOAM 1.38.0 has shipped to production.

# Verification steps
We need to verify that the RHOAM install and uninstall logic works for both fresh install and upgrade scenarios.

## Verify upgrade scenario
1. Checkout `rhoam-v1.38.0`
```bash
git checkout rhoam-v1.38.0
```
2. Prepare the cluster:
```bash
make cluster/prepare/local
```
3. Create the CatalogSource:
```bash
cat << EOF | oc create -f -
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: rhoam-operators
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/ckyrillo/managed-api-service-index:1.38.0
EOF
```
4. Create the RHOAM CR:
```bash
IN_PROW=false USE_CLUSTER_STORAGE=true make deploy/integreatly-rhmi-cr.yml
```
5. Login into the cluster and install RHOAM via OperatorHub
6. Verify that the installation completes without error
7. Update the CatalogSource to point to the upgrade image: `quay.io/ckyrillo/managed-api-service-index:1.39.0`
8. Verify that the upgrade completes without error
9. Clean up the cluster for the fresh install scenario
```bash
make cluster/cleanup
make cluster/cleanup/crds
```

## Verify fresh install scenario
1. Checkout this PR
2. Prepare the cluster:
```bash
make cluster/prepare/local
```
3. Create the CatalogSource:
```bash
cat << EOF | oc create -f -
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: rhoam-operators
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/ckyrillo/managed-api-service-index:1.40.0
EOF
```
4. Create the RHOAM CR:
```bash
IN_PROW=false USE_CLUSTER_STORAGE=true make deploy/integreatly-rhmi-cr.yml
```
5. Login into the cluster and install RHOAM via OperatorHub
6. Verify that the installation completes without error
